src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/heatmap.js

```diff
@@ -0,0 +1,203 @@
+/** Heatmap Visualization. */
+// TODO refactor this to avoid loading on startup. See issue #985 .
+loadScript('https://d3js.org/d3.v4.min.js')
+loadStyle('https://fontlibrary.org/face/dejavu-sans-mono')
+
+/**
+ * A d3.js heatmap visualization.
+ */
+class Heatmap extends Visualization {
+    static inputType = 'Any'
```

Comment:
This should be done more specifically. We don't want heatmap to be shown for example for strings, right? See other visualizations to see how to define it for a particular type or set of types.